lib/Backend/GlobOpt.cpp

```diff
@@ -10961,7 +10961,14 @@ GlobOpt::TypeSpecializeBinary(IR::Instr **pInstr, Value **pSrc1Val, Value **pSrc
 
             // Try to type specialize to int32
 
-            lossy = false;
+            // If one of the values is a float constant with a value that fits in a uint32 but not an int32, 
+            // and the instruction can ignore int overflow, the source value for the purposes of int specialization 
+            // would have been changed to an int constant value by ignoring overflow. But, the conversion is still lossy.
+            if (!(src1OriginalVal && src1OriginalVal->GetValueInfo()->IsFloatConstant() && src1Val && src1Val->GetValueInfo()->HasIntConstantValue()) &&
```

Comment:
Seems both src1Val and src2Val need to be checked. Pls ignore this comment.

---
In reply to: [58146247](https://github.com/Microsoft/ChakraCore/pull/700#discussion_r58146247) [](ancestors = 58146247)